### PR TITLE
feature(js): requiresConfirmation now returns false if not confirmed

### LINKED ISSUE #9240: js requiresConfirmation should also stop event propagation
otherwise `.on` registrations of other events on the same element will not be blocked

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #9240

### CODE REVIEW

File: js/lib/ui.js [hunk 1]
```diff
@@ -290,7 +290,7 @@ elgg.ui.initHoverMenu = function(parent) {
 };
 
 /**
- * Calls a confirm() and prevents default if denied.
+ * Calls a confirm() and returns false and prevents default if denied.
```
Comment (hypeJunction) [inline js/lib/ui.js]: too many ands. maybe break the line into to
if denied, returns ...

File: js/lib/ui.js [hunk 2]
```diff
@@ -299,6 +299,7 @@ elgg.ui.requiresConfirmation = function(e) {
 	var confirmText = $(this).data('confirm') || elgg.echo('question:areyousure');
 	if (!confirm(confirmText)) {
 		e.preventDefault();
```
Comment (hypeJunction) [inline js/lib/ui.js]: This will be irrelevant if you return false. It will prevent the default stop propagation/bubbling.

**Comment by hypeJunction** [comment]
@jdalsem Did this ever actually work for you? I am now adding a button with `data-confirm` and still fires my handler...

**Comment by jdalsem** [comment]
nope... not working for me.. do not know why previous tests did work .... it should work as described in #10034

**Comment by jdalsem** [comment]
i guess we can revert this change

**Pull request merged**